src/main.rs

```diff
@@ -1,124 +1,96 @@
-use std::env;
-use std::path::{Path, PathBuf};
 use std::time::Instant;
 
+use cli::{Cli, Command};
 use utils::net::{get_available_port, port_is_available};
 
+use clap::Parser;
+
 mod cli;
 mod cmd;
 mod console;
 mod prompt;
 
 fn main() {
-    let matches = cli::build_cli().get_matches();
-
-    let root_dir = match matches.value_of("root").unwrap() {
-        "." => env::current_dir().unwrap(),
-        path => PathBuf::from(path)
-            .canonicalize()
-            .unwrap_or_else(|_| panic!("Cannot find root directory: {}", path)),
-    };
-    let config_file = match matches.value_of("config") {
-        Some(path) => PathBuf::from(path)
-            .canonicalize()
-            .unwrap_or_else(|_| panic!("Cannot find config file: {}", path)),
-        None => root_dir.join("config.toml"),
-    };
+    let cli = Cli::parse();
+    let root_dir = cli
+        .root
+        .canonicalize()
+        .unwrap_or_else(|_| panic!("Cannot find root directory: {}", cli.root.display()));
+    let config_file = cli
+        .config
+        .map(|path| {
+            path.canonicalize()
+                .unwrap_or_else(|_| panic!("Cannot find config file: {}", path.display()))
+        })
+        .unwrap_or_else(|| root_dir.join("config.toml"));
 
-    match matches.subcommand() {
-        ("init", Some(matches)) => {
-            let force = matches.is_present("force");
-            match cmd::create_new_project(matches.value_of("name").unwrap(), force) {
-                Ok(()) => (),
-                Err(e) => {
-                    console::unravel_errors("Failed to create the project", &e);
-                    ::std::process::exit(1);
-                }
-            };
-        }
-        ("build", Some(matches)) => {
+    match cli.command {
+        Command::Init { name, force } => match cmd::create_new_project(&name, force) {
```

Comment:
looks like it would be easier to use `if let Err(e) = ...` than match

Reply:
True, I wanted to avoid doing refactorings unrelated strictly to the clap porting

Reply:
That would be a fine change if you don't mind